addons/mail/tests/discuss/test_discuss_sub_channels.py

```diff
@@ -31,26 +31,27 @@ def test_01_gc_unpin_outdated_sub_channels(self):
             self.env["discuss.channel.member"]._gc_unpin_outdated_sub_channels()
             self.assertTrue(self_member.is_pinned)
 
-    def test_02_sub_channel_members_sync_with_parent(self):
-        parent = self.env["discuss.channel"].create({"name": "General"})
-        parent.action_unfollow()
-        self.assertFalse(any(m.is_self for m in parent.channel_member_ids))
-        parent._create_sub_channel()
-        sub_channel = parent.sub_channel_ids[0]
-        # Member created for sub channel (_create_sub_channel): should also be
-        # created for the parent channel.
-        self.assertTrue(any(m.is_self for m in parent.channel_member_ids))
-        self.assertTrue(any(m.is_self for m in sub_channel.channel_member_ids))
+    def test_02_sub_channel_members(self):
+        bob_user = new_test_user(self.env, "bob_user", groups="base.group_user")
+        baz_user = new_test_user(self.env, "baz_user", groups="base.group_user")
+        parent = self.env["discuss.channel"].with_user(bob_user).create({"name": "General"})
+        self.assertIn(bob_user.partner_id, parent.channel_member_ids.partner_id)
+        parent.add_members(partner_ids=[baz_user.partner_id.id])
+        sub_channel = parent._create_sub_channel()
+        # Member created for sub channel (_create_sub_channel): should be only
+        # the sub channel creator.
+        self.assertIn(bob_user.partner_id, sub_channel.channel_member_ids.partner_id)
+        self.assertNotIn(baz_user.partner_id, sub_channel.channel_member_ids.partner_id)
```

Comment:
Why is this part changed actually?
The PR is only about fixing `unlink`, why is a test about "create" changed?

Reply:
The description in the test was `Member created for sub channel (_create_sub_channel): should also be created for the parent channel` and it was considering all members in the channel are adding to the sub channel too. The test was passed before because it was using only odoo bot as a member so in adding the sub channel it was only sub channel creator and checking if all members of parent are in the child was always correct.  In the first version of this PR I changed the code to support this behavior but as you said this is not what we want. I changed the test to focus on correct members in the parent that are not automatically added to the child. It could be in a separate PR, but since we had the related discussion here and the test was already changed, I kept it here.

Reply:
I don't think the test was covering that all members in parents are in child.

It was covering that adding member to child also added members to parent, which is necessary to pass ACL check of child (based on parent).

This is what the test should cover and it no longer covers that now.

Reply:
Is it green if you fully revert diff on this test?

Reply:
sync:
- create sub member -> create parent member
- unlink parent member -> unlink sub member
not sync:
- unlink sub member (parent member is not changed, just leaving sub is allowed)
- create parent member (sub member is not changed, just joining parent and not joining sub is allowed)

Reply:
and the coverage of this test (1st step) was:

- have empty channel (at least self not being member)
- create a sub channel (and therefore create a sub channel member)
- by sync mechanism: parent member should be created

Reply:
I reverted the diff on this test.